static/compiler-service.js

```diff
@@ -50,6 +51,73 @@ function CompilerService() {
     }, this);
 }
 
+CompilerService.prototype.getDefaultCompilerForLang = function (langId) {
+    return options.defaultCompiler[langId];
+};
+
+CompilerService.prototype.processFromLangAndCompiler = function (languageId, compilerId) {
+    var langId = languageId;
+    var foundCompiler = null;
+    try {
+        if (langId) {
+            var compilers = this.getCompilersForLang(langId);
+            if (!compilerId) {
+                compilerId = this.getDefaultCompilerForLang(langId);
+            }
+            var firstCompiler = null;
+            // We make sure we can find it (We might have been given a bad compiler)
+            foundCompiler = _.find(compilers, function (compiler) {
+                if (!firstCompiler) {
+                    firstCompiler = compiler;
+                }
+                return compiler.id === compilerId || compiler.alias === compilerId;
+            });
+            if (!foundCompiler) {
+                foundCompiler = firstCompiler;
+            }
+        } else if (compilerId) {
+            var languages = options.languages;
+            _.any(languages, function (lang) {
```

Comment:
Not a fan of abusing _.any for setting 2 vars outside of the any. Maybe a map that returns [{langId, foundCompiler} and the rest null] ? I believe _.compact can get rid of the nulls after that (or _.filter)

Reply:
realized that map causes unnecessary work, but not sure that codepath is visited a lot anyway